src/test/java/com/google/devtools/build/lib/actions/ActionInputMapTest.java

```diff
@@ -134,6 +134,10 @@ public void putTreeArtifact_addsTreeArtifactAndAllChildren() {
     assertContainsEqualMetadata(tree, treeValue.getMetadata());
     assertContainsSameInstance(child1, child1Metadata);
     assertContainsSameInstance(child2, child2Metadata);
+
+    assertThat(map.getMetadata(tree)).isEqualTo(treeValue.getMetadata());
+    assertThat(map.getTreeMetadata(tree.getExecPath())).isEqualTo(treeValue);
```

Comment:
Consider adding test cases specific to the new API rather than tagging along a single test. It's likely easy to parameterize as well e.g. tree with some files, empty tree etc.

Reply:
Done in 8d76ef1.